packages/nimble-components/src/wafer-map/index.ts

```diff
@@ -216,27 +219,28 @@ export class WaferMap<
         if (this.validity.invalidDiesTableSchema) {
             return;
         }
-
         if (this.waferMapUpdateTracker.requiresEventsUpdate) {
-            if (
-                this.waferMapUpdateTracker.requiresContainerDimensionsUpdate
-                || this.waferMapUpdateTracker.requiresScalesUpdate
-            ) {
-                this.experimentalDataManager.updateComputations();
-                await this.workerRenderer.setupWafer();
-                await this.workerRenderer.drawWafer();
+            let setupWafer = false;
+            if (this.waferMapUpdateTracker.requiresContainerDimensionsUpdate) {
+                this.computations.componentResizeUpdate();
+                setupWafer = true;
+            } else if (this.waferMapUpdateTracker.requiresScalesUpdate) {
+                this.computations.inputDataUpdate();
+                setupWafer = true;
             } else if (
                 this.waferMapUpdateTracker.requiresLabelsFontSizeUpdate
                 || this.waferMapUpdateTracker.requiresDiesRenderInfoUpdate
             ) {
-                this.experimentalDataManager.updatePrerendering();
-                await this.workerRenderer.drawWafer();
-            } else if (this.waferMapUpdateTracker.requiresDrawnWaferUpdate) {
-                await this.workerRenderer.drawWafer();
+                this.computations.colorAndTextUpdate();
+                setupWafer = true;
             }
-        } else if (this.waferMapUpdateTracker.requiresRenderHoverUpdate) {
-            this.workerRenderer.renderHover();
+            const snapshot = this.createSnapshot();
+            if (setupWafer) {
```

Comment:
Instead of tracking a separate flag here, is it possible to add something to the wafermaptracker like requiresWaferSetup?

something like
```
    public get requiresWaferSetup(): boolean {
        return (
            this.requiresContainerDimensionsUpdate
            || this.waferMapUpdateTracker.requiresScalesUpdate
            || this.waferMapUpdateTracker.requiresLabelsFontSizeUpdate
            || this.waferMapUpdateTracker.requiresDiesRenderInfoUpdate
        );
    }
```

The point of the wafermaptracker is to encapsulate that kind of conditional flag tracking logic.